src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -230,6 +245,56 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator {
             await RespondAsync(request, ct, input);
         }
 
+        public Task<SendBlobResult> SendBlobAsync(byte[] data, CancellationToken ct) {
```

Comment:
Under-the-hood difference is that for evaluation request requires only one websocket message,  whereas for blobs we need two websocket messages; text message [which has the header] followed by binary with the data. We did this to avoid duplicating the header parsing code (which we have for text) for binary.

After I make the change suggested by @int19h (see comment on SendBlobRequest.cs). There is  explicit value returned from send and get, instead of a result object which gets propagated to the caller. The caller does not get to choose how they want the result. That is different from evaluation result, where the result object makes sense. Since the result value depends on the expression and expectation of the caller.

We could create a base request type that manages the request id, and message name. We can then derive `EvaluationRequest` and `BlobRequest` from it, wherefore one dictionary should be enough to manage request and response agnostic of eval or blob.